apps/sms/services/js/moz_settings/moz_settings_shim.js

```diff
@@ -0,0 +1,65 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* global BridgeServiceMixin */
+
+/* exported MozSettingsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozSettings API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-settings-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = Object.freeze(['get', 'set']);
+
+var mozSettings = null;
+
+var MozSettingsShim = {
+  init(settings) {
+    if (!settings) {
+      return;
+    }
+    mozSettings = settings;
+
+    this.initService();
+  },
+
+  /* Methods */
+
+  /**
+   * Shim for mozSettings.createLock().get API.
+   * @param {String} key Key name for Settings.
+   * @returns {Promise} Promise that return the settings value from settings DB.
+   */
+  get(key) {
+    return mozSettings.createLock().get(key);
+  },
+
+  /**
+   * Shim for mozSettings.createLock().set API.
+   * @param {Object} settings Settings key and value for settings. It can have
+   *  several key/value pairs inside.
+   * @returns {Promise} Promise that return saved result.
+   */
+  set(settings) {
+    return mozSettings.createLock().set(settings);
+  }
+};
+
+exports.MozSettingsShim = Object.seal(
+  BridgeServiceMixin.mixin(
```

Comment:
Move the question in https://github.com/mozilla-b2g/gaia/pull/31003#discussion_r34971061 to here: Should we do so error handling in BridgeServiceMixin if we want to utilize it for shims.